### ISSUE TITLE
What is it for?

### ISSUE THREAD

**Issue by deathmood** [facebook/reason#528]
Maybe it is a dumb question, but I cant understand what is `Reason` for. There is a lot of info on the official website, but no answer to this question. So what can I do with `Reason`? What is the purpose of the language? What can I use it for? Why did you build one more programming language?

**Comment by jordwalke** [comment]
From the [main page](http://facebook.github.io/reason/):

> Reason is a new interface to OCaml - a highly expressive dialect of the ML language featuring type inference and static type checking. Reason provides a new syntax and toolchain for editing, building, and sharing code, and will evolve in the open as a community collaboration. Reason already has contributors across several organizations/companies.
Compare JS Compare ML

And then from the `README`, there are a few bullet points about (part of) the total developer experience you would receive which are reasons you'd want to use Reason. Numbers 1/2 provide a more approachable interface to 3/4 which are already provided by either the `ocamlopt` compiler, or one of the compile-to-js options that exist for the OCaml compiler. #5 is still WIP, and would provide a local development workflow for rapidly building and sharing packages.

1.  Approachable syntax.
2. Powerful, automatic source code formatting.
3. Adopt Incrementally with JavaScript/C interop.
4. Ahead of time compilation to assembly - without a language level VM.
5. Rapidly develop and share projects.

We've made good progress towards #1/#2 (it's in a usable state), and #5 is the biggest remaining challenge.

In summary, if you want many of the benefits of the OCaml compiler and the ones that are coming in the future, and you either want a more approachable syntax for yourself or the people who read your code, and you want it to be easy to develop many small connected packages locally, then Reason may be useful for you either now or some time in the future. If not, then Reason may not be useful for you.

**Comment by jordwalke** [comment]
Closing to keep clean out the task queue, but feel free to continue a discussion here if you like.
